client/src/components/AddQuestionForm.tsx

```diff
@@ -0,0 +1,156 @@
+import { useForm, SubmitHandler } from "react-hook-form";
+import { useState } from "react";
+import FormInputField from "./FormInputField";
+import Spinner from "./Spinner";
+import Button from "./Button";
+import * as adminQuickRound from "../newtork/adminQuickRound";
+import { useQueryClient } from "@tanstack/react-query";
+
+interface Answer {
+  answer: string;
+  numberOfResponse: number;
+}
+
+interface FormValues {
+  question: string;
+  answers: Answer[];
+}
+
+export default function AddQuestionForm({
+  closeModal,
+}: {
+  closeModal: () => void;
+}) {
+  const queryClient = useQueryClient();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<FormValues>();
+
+  const [numOfAnswers, setNumOfAnswers] = useState(3);
+  const [correctScore, setCorrectScore] = useState<boolean>(true);
+
+  const handleAnswerChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = parseInt(event.target.value, 10);
+    if (!isNaN(value) && value > 0) {
+      if (value > 10) {
+        setNumOfAnswers(10);
+      } else {
+        setNumOfAnswers(value);
+      }
+    }
+  };
+
+  interface AnswerSendProp {
+    [key: string]: number;
+  }
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    const filteredData = {
+      ...data,
+      answers: data.answers.filter(
+        (answer) =>
+          answer.answer.trim() !== "" && answer.numberOfResponse !== null
+      ),
+    };
+
+    const totalResponses = filteredData.answers.reduce(
+      (acc, answer) => acc + +answer.numberOfResponse,
+      0
+    );
+    if (totalResponses !== 100) {
+      console.error("Total responses must be 100");
+      setCorrectScore(false);
+      return;
+    }
+
+    const answersToSend: AnswerSendProp = {};
+    filteredData.answers.forEach((answer) => {
+      answersToSend[answer.answer] = +answer.numberOfResponse;
+      console.log("answersToSend", answersToSend);
```

Comment:
remove this console.log

Reply:
do i need to remove console.error as well?

Reply:
if the backend returns something sensitive, yes.